src/rgw/rgw_data_sync.cc

```diff
@@ -939,7 +939,7 @@ class RGWRunBucketSyncCoroutine : public RGWCoroutine {
     logger.init(sync_env, "Bucket", bs.get_key());
   }
   ~RGWRunBucketSyncCoroutine() override {
-    if (lease_cr) {
+    if (lease_cr.get()) {
```

Comment:
intrusive_ptr has conversion to bool, this can remain `if (lease_cr) {`. same with other if statements below